autowiring/AutoCurrentPacket.h

```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2012-2015 Leap Motion, Inc. All rights reserved.
+#pragma once
+#include "AutoPacket.h"
+
+/// <summary>
+/// RAII for AutoPacket::SetCurrent
+/// </summary>
+class AutoCurrentPacket
+{
+public:
+  AutoCurrentPacket(AutoPacket& apkt) {
+    AutoPacket::SetCurrent(apkt.shared_from_this());
+  };
+  ~AutoCurrentPacket(void) {
+    AutoPacket::SetCurrent(nullptr);
+  };
+};
```

Comment:
Can you change the name of this class to `AutoCurrentPacketPusher` and put it in the `autowiring` namespace?